addons/analytic/models/analytic_plan.py

```diff
@@ -244,38 +245,71 @@ def _get_applicability(self, **kwargs):
 
     def unlink(self):
         # Remove the dynamic field created with the plan (see `_inverse_name`)
+        if not self:
+            return super().unlink()
         self._find_plan_column().unlink()
         return super().unlink()
 
-    def _find_plan_column(self):
-        return self.env['ir.model.fields'].sudo().search([
-            ('name', 'in', [plan._strict_column_name() for plan in self]),
-            ('model', '=', 'account.analytic.line'),
-        ])
+    def _get_plan_column_models(self):
+        # Return the models on which the plan column should be created
+        return ['account.analytic.line']
+
+    def _find_plan_column(self, model=False):
+        domain = [('name', '=', self._strict_column_name())]
```

Comment:
Why transforming this into a function that only allows for one record in `self`?
If you don't make this change, you also don't need to change `unlink`...